packages/block-editor/src/components/inserter/media-tab/media-preview.js

```diff
@@ -121,9 +121,16 @@ export function MediaPreview( { media, onClick, category } ) {
 		useState( false );
 	const [ isHovered, setIsHovered ] = useState( false );
 	const [ isInserting, setIsInserting ] = useState( false );
+	const { getSettings } = useSelect( blockEditorStore );
+	const { imageDefaultSize } = getSettings();
```

Comment:
We shouldn't call the selector during the render. The actual setting value is only needed in the `onMediaInsert` callback. The `sizeSlug` does nothing for remote images.

P.S. I've got a small PR that improves the `onMediaInsert` callback and uses the same technique for the `mediaUpload` setting.